### DISCUSSION TITLE
stateless ViewStore in 1.0?

### DISCUSSION THREAD

**Discussion by jaredh159** [pointfreeco/swift-composable-architecture#2344] [Q&A] [answered]
was running a fairly recent version of the prerelease, but am trying to get on 1.0.0 today. got a compiler error where i was doing `ViewStore(store.stateless)`.  looks like the `stateless` and `actionless` computed properties on `Store` have been removed? grepped around for a bit and can't find it in the 1.0.0 codebase.

what's the recommended replacement? or is there some reason why i shouldn't be trying to do have a stateless view store anymore?

thanks!

**Comment by premyslvlcek** [comment, marked as answer]
Hi. Recommended replacement for `ViewStore(store.stateless)` is sending the actions directly to the store using `store.send(.action)`. It also returns StoreTask, so you can even write `MySwiftUIView.task { await store.send(.task).finish() }`

**Reply by jaredh159** [reply]
thanks!
